### PR TITLE
supporting partial in non-html context

### PR DESCRIPTION
**Pull request by neraliu**

### CODE REVIEW

File: src/express-secure-handlebars.js [hunk 1]
```diff
@@ -27,11 +43,108 @@ util.inherits(ExpressSecureHandlebars, expressHandlebars);
...
+        /* call the getTemplate() and create static/dynamic partials in order */ 
+        var expressSecureHandlebars = this;
+
+        /* return the Promise from render() */
+        return expressHandlebars.prototype.getTemplate.call(this, filePath, options).then(function(template) {
+            var compiledPartials = {};
+            function compilePartialCache(partialCache) {
+                for (var processedPartialName in partialCache) {
+                    if (partialCache.hasOwnProperty(processedPartialName)) {
+                        compiledPartials[processedPartialName] = precompiled? expressSecureHandlebars._precompileTemplat(partialCache[processedPartialName], expressSecureHandlebars.compilerOptions)
```
Comment (adon-at-work) [inline src/express-secure-handlebars.js]: what is `_precompileTemplat`?

Comment (neraliu) [inline src/express-secure-handlebars.js]: good catch, typo

File: src/express-secure-handlebars.js [hunk 2]
```diff
@@ -9,8 +9,12 @@ Authors: Nera Liu <neraliu@yahoo-inc.com>
 */
 /*jshint -W030 */
 var util = require("util"),
-    expressHandlebars = require('express-handlebars').ExpressHandlebars,
-    secureHandlebars = require('secure-handlebars');
+    ExpressHandlebars = require('express-handlebars').ExpressHandlebars,
+    secureHandlebars = require('secure-handlebars'),
```
Comment (adon-at-work) [inline src/express-secure-handlebars.js]: capitalize or use camel case for both?

File: src/express-secure-handlebars.js [hunk 3]
```diff
@@ -9,8 +9,12 @@ Authors: Nera Liu <neraliu@yahoo-inc.com>
...
+    assign = require('object.assign'),
+    Promise = require('bluebird'),
+    fs = Promise.promisifyAll(require("fs")),
+    glob = Promise.promisify(require('glob'));
```
Comment (adon-at-work) [inline src/express-secure-handlebars.js]: remove fs and glob, and dependencies at package.json

File: src/express-secure-handlebars.js [hunk 4]
```diff
@@ -20,18 +24,63 @@ function ExpressSecureHandlebars(config) {
...
     /* calling super constructor */
     this.constructor.super_.call(this, config);
+
+    // compilerOptions is being used for passing params to context-parser-handlebars
+    this.compilerOptions || (this.compilerOptions = {});
+    this.compilerOptions.cph || (this.compilerOptions.cph = {});
+    this.compilerOptions.cph.rawPartialsCache = {};
+    this.compilerOptions.cph.processedPartialsCache = {};
+    this.compilerOptions.cph.compiledPartialsCache = {};
+    this.compilerOptions.cph.enablePartialCombine = false;
```
Comment (adon-at-work) [inline src/express-secure-handlebars.js]: this gives no chance to set `compilerOptions.cph.property`

File: src/express-secure-handlebars.js [hunk 5]
```diff
@@ -20,18 +24,63 @@ function ExpressSecureHandlebars(config) {
...
+    this.slaveExpressHandlebars = new ExpressHandlebars(config);
+    var returnTemplate = function (template, options) {
+        return template;
+    };
+    this.slaveExpressHandlebars._compileTemplate = returnTemplate;
+    this.slaveExpressHandlebars._precompileTemplate = returnTemplate;
+    this.slaveExpressHandlebars.getPartials({}).then(function(rawPartialCache) {
+        expressSecureHandlebars.compilerOptions.cph.rawPartialsCache = rawPartialCache;
+        // disable the getPartials with partialsDir setting to empty array
+        expressSecureHandlebars.partialsDir = [];
```
Comment (adon-at-work) [inline src/express-secure-handlebars.js]: potential async issue.
move this line after line 38

File: src/express-secure-handlebars.js [hunk 6]
```diff
@@ -20,18 +24,63 @@ function ExpressSecureHandlebars(config) {
...
+            this.compilerOptions.cph.compiledPartialsCache[name] = precompiled? this._precompileTemplate(templates[name], this.compilerOptions) 
+                : this._compileTemplate(templates[name], this.compilerOptions);
+        }
+    }
+}
 
-/* override ExpressHandlebars.render() to expose filePath as compilerOptions */
+/* override ExpressHandlebars.render() */
 ExpressSecureHandlebars.prototype.render = function (filePath, context, options) {
+console.log("r:"+filePath);
```
Comment (adon-at-work) [inline src/express-secure-handlebars.js]: kill this

File: src/express-secure-handlebars.js [hunk 7]
```diff
@@ -20,18 +24,63 @@ function ExpressSecureHandlebars(config) {
...
+    // master express-secure-handlebars
+    var expressSecureHandlebars = this;
 
-    return expressHandlebars.prototype.render.call(this, filePath, context, options);
+    // return the Promise from render()
+    return ExpressHandlebars.prototype.getTemplate.call(this, filePath, options).then(function(template) {
+        compileTemplates.call(expressSecureHandlebars, expressSecureHandlebars.compilerOptions.cph.processedPartialsCache, options.precompiled);
+        // options.partials may be set in renderView, we concat it.
+        options.partials || (options.partials = {});
+        options.partials = assign({}, options.partials, expressSecureHandlebars.compilerOptions.cph.compiledPartialsCache);
```
Comment (adon-at-work) [inline src/express-secure-handlebars.js]: note the order

**Comment by adon-at-work** [comment]
the improvements have been made in https://github.com/yahoo/express-secure-handlebars/pull/24

**Pull request closed without merging**